### PR TITLE
Mark couple of helper functions in XdsClient as public. 

### PR DESCRIPTION
**Pull request by anicr7**
This will allow users who define and control custom xDS resources to reuse common methods.

@YifeiZhuang

### CODE REVIEW

**Review by YifeiZhuang — DISMISSED** [review]

File: xds/src/main/java/io/grpc/xds/XdsClient.java [hunk 1]
```diff
@@ -77,7 +77,10 @@ static boolean isResourceNameValid(String resourceName, String typeUrl) {
     return true;
   }
 
-  static String canonifyResourceName(String resourceName) {
+  /*
+   * Convert the XDSTP resource name to its canonical version.
+   */
+  public static String canonifyResourceName(String resourceName) {
```
Comment (YifeiZhuang) [inline xds/src/main/java/io/grpc/xds/XdsClient.java:83]: Can you merge from the master and mark this static method as experimental API. I am aware the class itself is marked as experimental too.
`@ExperimentalApi("https://github.com/grpc/grpc-java/issues/10862")`

File: xds/src/main/java/io/grpc/xds/XdsClient.java [hunk 2]
```diff
@@ -101,7 +104,10 @@ static String canonifyResourceName(String resourceName) {
     return resourceName.replace(rawQuery, canonifiedQuery);
   }
 
-  static String percentEncodePath(String input) {
+  /*
+   * Percent encode the input using the url path segment escaper.
+   */
+  public static String percentEncodePath(String input) {
```
Comment (YifeiZhuang) [inline xds/src/main/java/io/grpc/xds/XdsClient.java:110]: same here, can you add
`@ExperimentalApi("https://github.com/grpc/grpc-java/issues/10862")`

File: xds/src/main/java/io/grpc/xds/XdsClient.java (hunk 1 above)
Comment (anicr7) [reply xds/src/main/java/io/grpc/xds/XdsClient.java:86]: Done.

File: xds/src/main/java/io/grpc/xds/XdsClient.java (hunk 2 above)
Comment (anicr7) [reply xds/src/main/java/io/grpc/xds/XdsClient.java:114]: Done

**Review by YifeiZhuang — APPROVED** [review]

**Comment by ejona86** [comment]
These API changes are not for external use. We should really mark them `@Internal`. Although I guess we could fix that stuff when we move xds client to its own package.

**Pull request merged**